src/main/java/org/sonarsource/sonarlint/ls/settings/SettingsManager.java

```diff
@@ -228,6 +228,7 @@ CompletableFuture<Map<String, Object>> requestSonarLintAndOmnisharpConfiguration
 
   static Map<String, Object> updateAnalyzerProperties(@org.jetbrains.annotations.Nullable URI workspaceUri, List<Object> response, Map<String, Object> settingsMap) {
     var analyzerProperties = (Map<String, String>) settingsMap.getOrDefault(ANALYZER_PROPERTIES, Maps.newHashMap());
+    analyzerProperties.put("sonar.cs.file.suffixes", ".cs");
```

Comment:
I would maybe extract this into an appropriately named method, e.g `forceIgnoreRazorFiles`.

Also, if the property is set, it should not be blindly overwritten - it can be different in the server settings.

I believe a more complete implementation would be to:
* check if the key is there
* if it is there and it contains `.razor`, split on `,`, remove `.razor` and `join` it back